docs/dependency-management.md

```diff
@@ -0,0 +1,44 @@
+# Dependency Management Policy for InstructLab
+
+This document describes the policies for adding and updating build and runtime dependencies of all InstructLab components.
+
+## Best practices
+
+<!-- This section modified from https://github.com/instructlab/training/issues/34#issuecomment-2176071856 -->
+1. Express dependencies by setting a minimum version (using `>=`) to ensure compatibility.
+2. Do not "pin" a dependency to a single version (using `==`).
+3. Exclude specific versions known to not work (using `!=`).
+4. Only apply "caps" to dependencies (using `<`) when that dependency has established a pattern of producing new releases with breaking changes.
+5. Pin versions in CI, with frequent automated updates.
+
+Best practices for python dependencies call for using ranges in package requirements and pinning versions only in CI jobs.
+
+Using pinned versions in, for example, `requirements.txt` or `constraints.txt`, allows us to know and advertise exactly what versions have been tested in CI.
+That information is useful for users and re-packagers to understand which versions of dependencies are compatible with more specificity than the ranges provide.
+Tools like Dependabot will submit PRs to automatically update those pins to help us keep up with new releases of all of our dependencies.
+
+Pinning to specific versions in the package dependencies in `pyproject.toml` or `requirements.txt` is not a good practice.
```

Comment:
in line 16, it suggests to use pinned versions in requirements.txt for CI, but here it suggests using pinned versions in requirement.txt is not a good practice. Just found this confusing.

Reply:
You don't want to pin your package dependencies. You do want to pin your CI packages. That's done in separate files. Both tend to be named requirements.txt. The earlier text had said `requirements.in` but that was changed to `requirements.txt`.

I don't know if you're using separate requirements/constraints lists for CI, yet.

Reply:
I made that suggestion assuming it was a typo - if not feel free to change it back

Reply:
The point of the suggestions is getting lost in the concern about specific filenames, so I've dropped the filenames and focused on how the requirements are used in different cases.

If this part is still confusing, I can just remove it all. The most important part for me right now is that the team understands their responsibility for keeping minimum versions up to date.